### DISCUSSION TITLE
Using LaTeX with `docker`

### DISCUSSION THREAD

**Discussion by hoelzer** [roaldarbol/LaPreprint#17] [Ideas]
Hey, thanks for the template!

> ! LaTeX Error: File `orcidlink.sty' not found.

I think I can [download the style file](https://github.com/duetosymmetry/orcidlink-LaTeX-command/blob/master/orcidlink.sty) and put it in the folder - but maybe it makes sense to put it in the repository as well. 

```
# this works for me to compile the pdf
wget https://raw.githubusercontent.com/duetosymmetry/orcidlink-LaTeX-command/master/orcidlink.sty .
pdflatex main.tex
```

**Comment by roaldarbol** [comment]
Hi @hoelzer! `orcidlink.sty` is now included in the TexLive/MikTeX distributions, so try updating your installation depending on how you're using LaTeX. It was only added recently, so it might not be part of your current installation. Let me know if that's somewhat on point.

**Comment by hoelzer** [comment]
Ah, I see! Okay, I don't have `latex` installed on my system and just use a `docker` container:

```
docker run --rm -v $PWD:$PWD -w $PWD nanozoo/pdflatex:3.14159265--6263fbd pdflatex main.tex
```

After manually downloading the `orcidlink.sty` this works for me.

I think compiling the PDF via a docker container might be also helpful for others (keeps the system clean, or if someone does not want to install TexLive/MikeTeX for any reason). 

I will try to add the missing style file to that `docker` container. If this works, I let you know here and then, if you want, you can also provide that command as an option to easily compile the template w/o any need of installing additional software (except `docker` ;) ).

**Comment by roaldarbol** [comment]
Oh, cool! I did previously also have the file as part of the repo, but I'm trying to keep it as clean as possible for now. If you'd be keen to do a short write-up about how to work with LaTeX (and the template) with Docker, I can add it to the wiki. 😃

**Reply by roaldarbol** [reply]
@hoelzer just moved the issue into a discussion instead. 😉

**Reply by hoelzer** [reply]
Good idea ;) sure! Let me update the docker and then I can provide smt you can add to the readme

**Comment by hoelzer** [comment]
@roaldarbol I updated the Docker image and now it comes w/ the necessary style file for ORCID. Proposed text for README.md if you like to add it:

---

## Compile PDF via Docker

If you don't have LaTeX installed, you can use a Docker image to compile a PDF from the LaTeX template. You only need [Docker installed](https://docs.docker.com/engine/install/) - nothing more. 

When you have Docker, clone this repository, change to the cloned directory, and run:

```bash
docker run --rm -v $PWD:$PWD -w $PWD nanozoo/pdflatex:3.14159265--f2f4a3f pdflatex main.tex
```

If you run the command for the first time, the Docker image will be automatically downloaded and then executed to compile the PDF.

**Reply by roaldarbol** [reply]
@hoelzer That's really awesome, thanks! I'll add it to the wiki - but I think that entire wiki entry will need a re-write to address the pluses and minuses for the different approaches, as well as a guide on how to get started with LaTeX altogether. But I think that's really positive as it means there's people from all corners of experience getting excited about nicely typeset papers!

**Reply by hoelzer** [reply]
Pleasure! And yes, that's really nice and powerful when people with different expertise contribute. Thanks again for making that public! I had also really good experiences with a LaTeX template for a german funding agency (DFG) I put public and which really improved due to help from the community (https://github.com/hoelzer/dfg, if interested)
